components/google_calendar/google_calendar.app.js

```diff
@@ -1,154 +1,307 @@
+      const resp = await calendar.calendarList.list();
+      return resp;
     },
     async list(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.events.list(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.events.list(config);
+      return resp;
     },
     // for config key value pairs - https://developers.google.com/calendar/v3/reference/events/list
     // deprecated
     async getEvents(config) {
-      return await this.list(config)
+      return await this.list(config);
     },
     async watch(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.events.watch(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.events.watch(config);
+      return resp;
     },
     async stop(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.channels.stop(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.channels.stop(config);
+      return resp;
     },
     async fullSync(calendarId) {
-      let nextSyncToken = null
-      let nextPageToken = null
-      while(!nextSyncToken) {
+      let nextSyncToken = null;
+      let nextPageToken = null;
+      while (!nextSyncToken) {
         const listConfig = {
           calendarId,
-          pageToken: nextPageToken
-        }
-        const syncResp = await this.list(listConfig)
-        nextPageToken = get(syncResp, "data.nextPageToken")
-        nextSyncToken = get(syncResp, "data.nextSyncToken")
+          pageToken: nextPageToken,
+        };
+        const syncResp = await this.list(listConfig);
+        nextPageToken = get(syncResp, "data.nextPageToken");
+        nextSyncToken = get(syncResp, "data.nextSyncToken");
       }
-      return nextSyncToken
-    }
-  }
-}
+      return nextSyncToken;
+    },
+    /**
+     * @param {string} [pageToken] - the page token for the next page of shared
+     * calendars
+     * @returns
+     */
+    async listCalendarsInPage(pageToken) {
+      const calendar = this.calendar();
+      const { data } = await calendar.calendarList.list({
+        pageToken,
+        showHidden: true,
+      });
+      return data;
+    },
+
+    /**
+     * @param {string} [calendarId] - user calender id
+     * @returns
+     */
+    async listEventOptions(calendarId) {
+      const calendar = this.calendar();
+      const { data } = await calendar.events.list({
+        calendarId,
+      });
+
+      const options = data.items.map((event) => {
+        return {
+          label: event.summary,
+          value: event.id,
+        };
+      });
+
+      return options;
+    },
+
+    /**
+     * @param {string} [calendarId] - User Calendar Id
+     * @returns
+     */
+
+    async listACLOptions(calendarId) {
+      const calendar = this.calendar();
+      const { data } = await calendar.acl.list({
```

Comment:
I would recommend creating a new function named `listAcls` like this
```javascript
    async listAcls(params) {
      const calendar = this.calendar();
      const { data } = await calendar.acl.list(params);
      return data;
    },
```
following the same pattern as the other functions